core-rust/state-manager/src/state_manager.rs

```diff
@@ -1005,26 +1001,27 @@ where
 
             let commit_result = match &processed.receipt().result {
                 TransactionResult::Commit(result) => {
-                    if let Some((_, next_epoch)) = result.next_epoch {
+                    if result.next_epoch.is_some() {
                         let is_last = i == (commit_transactions_len - 1);
                         if !is_last {
                             return Err(CommitError::MissingEpochProof);
                         }
-                        // TODO: Use actual result and verify proof validator set matches transaction receipt validator set
-                        epoch_boundary = Some(next_epoch);
+                        // TODO: verify that `result.next_epoch == commit_ledger_header.next_epoch`
```

Comment:
_(note: I have left the `TODO` because it was here before, but actually I don't think we have to explicitly check that `result.next_epoch` is same as expected - we will soon be checking that the entire receipt is same as expected [when I add the `transaction_receipt_root_hash` to the ledger header], which also includes the `next_epoch` anyways; do I get it right?)_

Reply:
yup, receipt verification should cover that